Due to some GIL and threading changes in the Python 2.7, cffi threads can break.

upstream bug:
https://bitbucket.org/cffi/cffi/issues/358/segfaults-in-testing-embedding

--- cffi-1.11.4/cffi/_embedding.h
+++ cffi-1.11.4/cffi/_embedding.h
@@ -332,8 +332,7 @@ static int _cffi_carefully_make_gil(void
     }
 #endif
 
-#if PY_MAJOR_VERSION >= 3
-    /* Python 3: call Py_InitializeEx() */
+    /* call Py_InitializeEx() */
     {
         PyGILState_STATE state = PyGILState_UNLOCKED;
         if (!Py_IsInitialized())
@@ -344,17 +343,6 @@ static int _cffi_carefully_make_gil(void
         PyEval_InitThreads();
         PyGILState_Release(state);
     }
-#else
-    /* Python 2: call PyEval_InitThreads() */
-# ifdef WITH_THREAD
-    if (!PyEval_ThreadsInitialized()) {
-        PyEval_InitThreads();    /* makes the GIL */
-        PyEval_ReleaseLock();    /* then release it */
-    }
-    /* else: there is already a GIL, but we still needed to do the
-       spinlock dance to make sure that we see it as fully ready */
-# endif
-#endif
 
 #ifdef WITH_THREAD
     /* release the lock */
